tests/test_client/tests.py

```diff
@@ -92,17 +93,46 @@ def test_post(self):
         self.assertEqual(response.templates[0].name, 'POST Template')
         self.assertContains(response, 'Data received')
 
-    def test_json_serialization(self):
-        """The test client serializes JSON data."""
+    def test_python_type_json_serialization(self):
+        """The test client serializes Python types to supported JSON types."""
         methods = ('post', 'put', 'patch', 'delete')
+
+        class IntEnum(enum.IntEnum):
+            IntValue = 1
+
+        python_types_expected_values = {
+            'dict': ({'value': 37}, {'value': 37}),
+            'list': ([37], [37]),
+            'tuple': ((37,), [37]),
+            'str': ("Value", "Value"),
+            'int': (37, 37),
+            'float': (37.0, 37.0),
+            'int_enum': (IntEnum.IntValue, 1),
```

Comment:
I think a few basic types are enough to test. This isn't testing anything particular in Django as far as I can tell.

Reply:
Fair enough. Given that they are there I think we might as well keep them. It makes it less likely that someone would try to guess all the types for an `isinstance` check.